lib/shoryuken/queue.rb

```diff
@@ -58,33 +68,48 @@ def sanitize_messages!(options)
       options
     end
 
+    def add_fifo_attributes!(options)
+      return unless fifo?
+
+      options[:message_group_id]         ||= MESSAGE_GROUP_ID
+      options[:message_deduplication_id] ||= Digest::SHA256.digest(options[:message_body])
+
+      options
+    end
+
     def sanitize_message!(options)
-      options = case
-                when options.is_a?(String)
-                  # send_message('message')
-                  { message_body: options }
-                when options.is_a?(Hash)
-                  options
-                end
+      options = { message_body: options } if options.is_a?(String)
 
+      add_fifo_attributes!(options)
       validate_message!(options)
 
       options
     end
 
     def validate_messages!(options)
-      options[:entries].map { |m| validate_message!(m) }
+      options[:entries].map(&method(:validate_message!))
     end
 
     def validate_message!(options)
       body = options[:message_body]
+
       if body.is_a?(Hash)
         options[:message_body] = JSON.dump(body)
       elsif !body.is_a?(String)
-        fail ArgumentError, "The message body must be a String and you passed a #{body.class}"
+        fail ArgumentError, "The message body must be a String and you passed a #{body.class}."
       end
 
+      validate_fifo_message!(options)
+
       options
     end
+
+    def validate_fifo_message!(options)
+      return unless fifo?
+
+      if options[:delay_seconds]
+        fail ArgumentError, 'FIFO queues do not accept DelaySeconds arguments.'
+      end
```

Comment:
@richseviora I removed other validation, because [this code](https://github.com/phstc/shoryuken/pull/276/files#r90748950) will prevent that to be `nil` anyways. WDYT?